### PR TITLE
feat: separate updatecli to its own pipeline

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: jenkins-infra/helpdesk#2778

(Reopening https://github.com/jenkins-infra/kubernetes-management/pull/2817 after https://github.com/jenkins-infra/kubernetes-management/pull/2865)

### CODE REVIEW

File: updatecli/updatecli.d/pod-templates/helmfile.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ targets:
     name: Update docker-helmfile in Jenkinsfile
     kind: file
     spec:
-      file: Jenkinsfile_k8s
+      file: Jenkinsfile_updatecli
```
Comment (dduportal) [inline updatecli/updatecli.d/pod-templates/helmfile.yaml:57]:
```suggestion
      files: 
      - Jenkinsfile_k8s
      - Jenkinsfile_updatecli
```

Don't we need the update on both pipelines?

Comment (lemeurherve) [reply updatecli/updatecli.d/pod-templates/helmfile.yaml:57]: No, the `jenkinsciinfra/helmfile` image is defined only in the updatecli stages, no mention of it in the remaining Jenkinsfile_k8s

Comment (lemeurherve) [reply updatecli/updatecli.d/pod-templates/helmfile.yaml:57]: As a matter of fact, this image shouldn't be defined in either of them as it's already the default included image specified and tracked in the shared pipeline library:
https://github.com/jenkins-infra/pipeline-library/blob/master/vars/updatecli.groovy#L10

One less dependencies location to deal with! Incoming commit.

Comment (dduportal) [reply updatecli/updatecli.d/pod-templates/helmfile.yaml]: You are correct: I missed the fact that it's centralized in  the `PodTemplate.yaml` file \o/

**Pull request merged**